internal/http/http.go

```diff
@@ -28,3 +32,31 @@ func SetCacheHeaders(w http.ResponseWriter, maxAge time.Duration, timeUntilExpir
 	w.Header().Set("Cache-Control", "public, max-age="+maxAge.String())
 	w.Header().Set("Expires", time.Now().Add(timeUntilExpiry).Format(time.RFC1123))
 }
+
+func LimitRead(r io.Reader) ([]byte, int, error) {
```

Comment:
Let's pass maxBytes as an arg so that we can set it differently on a case-by-case basis.

Reply:
Instead of returning a status code as an int, couldn't errors returned from this function be a `httpError` with the status code already set? Then the caller doesn't even need to think about looking at the status code or anything and can just send it directly to `WriteErrorJSON()`, which will know what to do with it?